lock-api/src/test/java/com/palantir/lock/watch/LockWatchEventTrackerImplTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static com.palantir.lock.watch.LockWatchInfo.State.LOCKED;
+import static com.palantir.lock.watch.LockWatchInfo.State.UNLOCKED;
+
+import java.util.OptionalLong;
+import java.util.UUID;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.lock.AtlasCellLockDescriptor;
+import com.palantir.lock.LockDescriptor;
+
+public class LockWatchEventTrackerImplTest {
+    private static final String TABLE_NAME = "test.table";
+    private static final byte[] ROW = new byte[] { 'r', 'o', 'w' };
+    private static final byte[] COL = new byte[] { 'c', 'o', 'l' };
+    private static final byte[] LOL = new byte[] { 'l', 'o', 'l' };
+    private static final LockDescriptor DESCRIPTOR = AtlasCellLockDescriptor.of(TABLE_NAME, ROW, COL);
+    private static final LockDescriptor OTHER_DESCRIPTOR = AtlasCellLockDescriptor.of(TABLE_NAME, ROW, LOL);
+    private static final LockWatchRequest ROW_PREFIX_REQ = LockWatchRequest.of(
+            ImmutableSet.of(LockWatchReferences.rowPrefix(TABLE_NAME, ROW)));
+
+    private final LockWatchEventTracker tracker = new LockWatchEventTrackerImpl();
+
+    @Test
+    public void stateWithNoUpdatesIsNotWatched() {
+        VersionedLockWatchState state = tracker.currentState();
+        assertThat(state.version()).isEmpty();
+        assertThat(state.lockWatchState(DESCRIPTOR)).isEqualTo(LockWatchInfo.NOT_WATCHED);
+    }
+
+    @Test
+    public void lockWatchCreatedWithoutOpenLockGivesNoInformationOnLocks() {
+        UUID leaderId = UUID.randomUUID();
+        LockWatchStateUpdate update = LockWatchStateUpdate.update(leaderId, 0, ImmutableList.of(
+                LockWatchCreatedEvent.builder(ROW_PREFIX_REQ, UUID.randomUUID()).build(0)));
+
+        tracker.updateState(update);
+        VersionedLockWatchState state = tracker.currentState();
+        assertThat(state.version()).isEqualTo(OptionalLong.of(0));
```

Comment:
```suggestion
        assertThat(state.version()).hasValue(0);
```

And so on, throughout the test class.

Reply:
I knew there must be a better way :(